### PR TITLE
simplified concepts in "possible architectures" section

### PR DESCRIPTION
**Pull request by juanmaguitar**

### CODE REVIEW

**Comment by juanmaguitar** [comment]
This PR includes several changes including:

- The separation of "Installation & Deploy" into "Architecture" & "Deployment"
- Reviewed "Deploy in now"
- Added "Deploy in heroku"
- Simplification of architecture (removed all non fully functional approaches)

Please, @luisherranz  can you have a look at it to check I'm not saying anything wrong

File: deployment/deploy-on-now.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+- configure your Frontity app to point to that custom domain 
+- configure your custom domain in the ZEIT server
+- add proper registries in your domain provider so the DNS communication is properly done
+
+...before deploying it
+
+### Add an `alias` in your [`now.json`](https://zeit.co/docs/configuration?query=now.json#introduction/configuration-reference)
+
+1. Add the alias field (`"www.your-site.com"`) with your custom domain in your `now.json`.
```
Comment (luisherranz) [inline deployment/deploy-on-now.md:58]: I think the alias field was deprecated in the last CLI update. They now ask you about that a prompt on your first deploy, don't they?

**Comment by juanmaguitar** [comment]
> The only thing that feels weird to me is that you have to add the /build folder to git in Heroku. I don't know why that is happening but feels wrong to me. Heroku should build that folder in the buildpack.

@luisherranz you're right that is not ideal but it's the best solution I've found in terms of easiness and performance

Some other solutions would be:

### Adding a `"prestart": "frontity build"` 

This would create a build every time the server is started but this increases a lot the loading time and it doesn't seem  like a solid solution (it fails sometimes)

It can be solved somehow by increasing the boot time from https://tools.heroku.support/limits/boot_timeout but it also fails sometimes with this

### Creating a custom [`buildpack`](https://devcenter.heroku.com/articles/buildpacks)

This seems to be like a good solution but I need to do some more research. For what I've seen I think it would imply some development similar to https://github.com/frontity/now-builder

File: deployment/deploy-on-now.md (hunk 1 above)
Comment (juanmaguitar) [reply deployment/deploy-on-now.md]: You're right. I have completely rewritten the "deploy to a custom domain" section

**Comment by mburridge** [comment]
I'm wondering why my changes from https://github.com/frontity/gitbook-docs/pull/9/commits/c0fc616bcaee0b3c605952f14898065420e6169f
are not incorporated in the later changes by @juanmaguitar at https://github.com/frontity/gitbook-docs/pull/9/commits/bfaf1af8d9c960aa343aba5765f2c3c1037f5559?

**Comment by mburridge** [comment]
Or even the most recent change at https://github.com/frontity/gitbook-docs/pull/9/commits/5c04ce73ea74cbc8ce2e50d0176fd87f49c8e0b5

**Comment by juanmaguitar** [comment]
I think it's my fault because when moving files at some point I took as a reference the current version of some files without considering changes you may have done
Your changes should be incorporated at 
[51f5fa8](https://github.com/frontity/gitbook-docs/pull/9/commits/51f5fa8b2a1702ef66d862b17112616c46c4e86d)

File: deployment/deploy-on-now.md (hunk 1 above)
Comment (luisherranz) [reply deployment/deploy-on-now.md]: Awesome 🙂

**Comment by juanmaguitar** [comment]
> I'm pretty sure there has to be a way to do a build in heroku without creating a custom buildpack. It's just so common.
> 
> From a quick look at the docs it seems like you should be able to use the `heroku-postbuild` script:
> https://devcenter.heroku.com/articles/nodejs-support#heroku-specific-build-steps
> 
> Could you please check it out?

@luisherranz I've reviewed and you're right. Heruku takes care of the `build` script and there's no need of "uploading" the `build` folder. Removed that block section from the instructions

**Comment by luisherranz** [comment]
Oh, they have Fastly and Cloudfront as add-ons:
- https://elements.heroku.com/addons/fastly
- https://elements.heroku.com/addons/edge

Maybe we can test them and recommend the best one.

If those services require `cache-control` headers in Frontity let me know.

**Comment by juanmaguitar** [comment]
@luisherranz no worries. I've moved the specific instructions about how to deploy to heroku to the PR #26  so we can do all the tests we need it before publishing the instructions

**Pull request merged**